### PR TITLE
Add documentation for Git branching strategy

### PR DESCRIPTION
**Pull request by stevenadams**
The purpose to this documentation is to add governance to how
contributors branch from master to make changes, how they name these
branches, and how to create a release of the repo

### CODE REVIEW

File: docs/contributing/git-branching-strategy.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#Git branching strategy
+**Author**: Steven Adams  
+**Date**: 9/10/17  
+
+For the nhsuk-prototype-kit repo we use the [MasterFlow branching strategy](https://medium.com/kainos/masterflow-simple-git-branching-e9b6844d64af). Branch names must conform to the following naming convention, where branch-type can be `feature`, `hotfix-[release-being-hotfixed]`, or `refactor`:
```
Comment (visualunderground) [inline docs/contributing/git-branching-strategy.md]: Might need an explanation of "refactor". I'm assuming "same on the outside, better on the inside" applies here.

Comment (visualunderground) [reply docs/contributing/git-branching-strategy.md]: If a typo was found in the docs, would this be a "hotfix-"?

**Review by visualunderground — APPROVED** [review]

**Comment by ghost** [comment]
Looks good. Any thoughts on strategy for release numbering? [semver](http://semver.org) is popular and well-regarded.

File: docs/contributing/git-branching-strategy.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Hotfixes
+A hotfix of a release can come in two forms, critical and non-critical. Non-critical hotfixes are issues that need fixed but are not considered show stoppers and can wait until the next scheduled release. Critical hotfixes are issues considered to be more urgent that justify a new "patched" release being published independently of the next release.
+
+In either case (critical or non-critical) create a branch:
+
+**Example**:
+`git checkout -b hotfix-v2.1/NDPE-24-hotfix-for-button-ui`
+
+Then make your changes and commit. Once the hotfix is complete, push your hotfix branch to origin, tag it with the relevant patch number (in the above example v2.1.1) and submit a merge request to master.
```
Comment (ghost) [inline docs/contributing/git-branching-strategy.md:25]: s/merge request/pull request/

**Review by ghost — APPROVED** [review]

File: docs/contributing/git-branching-strategy.md (hunk 1 above)
Comment (stevenadams) [reply docs/contributing/git-branching-strategy.md]: @visualunderground the refactor bit shouldm't have been left in. Good spot.

**Pull request merged**